app/src/main/java/com/kickstarter/ui/activities/DeepLinkActivity.java

```diff
@@ -0,0 +1,79 @@
+package com.kickstarter.ui.activities;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.kickstarter.libs.BaseActivity;
+import com.kickstarter.libs.RefTag;
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel;
+import com.kickstarter.libs.utils.ApplicationUtils;
+import com.kickstarter.ui.IntentKey;
+import com.kickstarter.viewmodels.DeepLinkViewModel;
+
+import java.util.List;
+
+import static com.kickstarter.libs.rx.transformers.Transformers.observeForUI;
+
+@RequiresActivityViewModel(DeepLinkViewModel.ViewModel.class)
+public class DeepLinkActivity extends BaseActivity<DeepLinkViewModel.ViewModel> {
+  @Override
+  protected void onCreate(@Nullable final Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    this.viewModel.outputs.startDiscoveryActivity()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(__ -> startDiscoveryActivity());
+
+    this.viewModel.outputs.startProjectActivity()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::startProjectActivity);
+
+    this.viewModel.outputs.startBrowser()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::startBrowser);
+
+    this.viewModel.outputs.requestPackageManager()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::inputPackageManager);
+  }
+
+  private void startDiscoveryActivity() {
+    ApplicationUtils.startNewDiscoveryActivity(this);
+    finish();
+  }
+
+  private void startProjectActivity(final String url) {
+    final Uri uri = Uri.parse(url);
```

Comment:
alternatively, we could output the Uri rather than Url since we only need the Uri to start the activity